doc/reference/networking/can_api.rst

```diff
@@ -145,14 +154,13 @@ a mailbox. When a transmitting mailbox is assigned, sending cannot be canceled.
   struct zcan_frame frame = {
           .id_type = CAN_STANDARD_IDENTIFIER,
           .rtr = CAN_DATAFRAME,
-          .ext_id = 0x123,
-          .dlc = 8
+          .id = 0x123,
+          .dlc = 8,
+          .data = {1,2,3,4,5,6,7,8}
   };
-  struct device *can_dev;
+  const struct device *can_dev;
   int ret;
 
-  frame.data = {1,2,3,4,5,6,7,8};
-
   can_dev = device_get_binding("CAN_0");
```

Comment:
```suggestion
  can_dev = device_get_binding(DT_CHOSEN_ZEPHYR_CAN_PRIMARY_LABEL);
```

Reply:
The code snipped should be as simple as possible. No macro magic.